test/stepper.js

```diff
@@ -62,6 +62,349 @@ exports["Stepper Firmware Requirement"] = {
+      stepper.pins,
+      {motor1: pins[0], motor2: pins[1], motor3: pins[2], motor4: pins[3]}
+    );
+
+    test.done();
+  },
+};
+
+exports["Stepper - max steppers"] = {
+  setUp: function(done) {
+
+    this.maxSteppers = 6;
+
+    this.board1 = new Board({
+      io: new MockFirmata({
+        pins: [
+          {
+            supportedModes: [8],
+          },
+        ]
+      }),
+      debug: false,
+      repl: false
+    });
+
+    this.board2 = new Board({
+      io: new MockFirmata({
+        pins: [
+          {
+            supportedModes: [8],
+          },
+        ]
+      }),
+      debug: false,
+      repl: false
+    });
+
+    done();
+  },
+
+  tearDown: function(done) {
+    done();
+  },
+
+  singleBoard: function(test) {
+    test.expect(2);
+
+    test.doesNotThrow(function() {
+      for (var i = 0; i < this.maxSteppers; i++) {
+        new Stepper({
+          board: this.board1,
+          stepsPerRev: 200,
+          pins: [3, 4, 5, 6]
+        });
+      }
+    }.bind(this));
+
+    test.throws(function() {
+      new Stepper({
+        board: this.board1,
+        stepsPerRev: 200,
+        pins: [3, 4, 5, 6]
+      });
+    }.bind(this));
+
+    test.done();
+  },
+
+  multipleBoards: function(test) {
+    test.expect(3);
+
+    // should be able to add max on two boards
+    test.doesNotThrow(function() {
+      for (var i = 0; i < this.maxSteppers; i++) {
+        new Stepper({
+          board: this.board1,
+          stepsPerRev: 200,
+          pins: [2, 3, 4, 5]
+        });
+        new Stepper({
+          board: this.board2,
+          stepsPerRev: 200,
+          pins: [2, 3, 4, 5]
+        });
+      }
+    }.bind(this));
+
+    // but can't add any more to either board
+    test.throws(function() {
+      new Stepper({
+        board: this.board1,
+        stepsPerRev: 200,
+        pins: [2, 3, 4, 5]
+      });
+    }.bind(this));
+    test.throws(function() {
+      new Stepper({
+        board: this.board2,
+        stepsPerRev: 200,
+        pins: [2, 3, 4, 5]
+      });
+    }.bind(this));
+
+    test.done();
+  }
+};
+
+// TODO test that callback is called for .step()
+// mock firmata apparently never calls it, so we can't test currently
+// exports["Stepper - step callback"] = {};
+
+exports["Stepper - set direction required before step"] = {
+  setUp: function(done) {
+
+    this.board = new Board({
+      io: new MockFirmata({
+        pins: [
+          {
+            supportedModes: [8],
+          },
+        ]
+      }),
+      debug: false,
+      repl: false
+    });
+
+    this.stepper = new Stepper({
+      board: this.board,
+      type: five.Stepper.TYPE.DRIVER,
+      stepsPerRev: 200,
+      pins: [2, 3]
+    });
+    done();
+  },
+
+  tearDown: function(done) {
+    done();
+  },
+
+  directionSet: function(test) {
+    test.expect(1);
+
+    // If we haven't set a direction, we should get error
+    this.stepper.step(1, function(error) {
+      test.ok(!!error);
```

Comment:
@scottgonzalez when using that `error.code` technique, instead of using `test.throws` do you just use a try/catch and then `test.equals` to check the code value? Or am I missing a cleaner method?